### DISCUSSION TITLE
On loading Sponza

### DISCUSSION THREAD

**Discussion by Hellwoodoo** [lwjglgamedev/vulkanbook#93] [Q&A]
Hi @lwjglgamedev, have been going through this wonderful resource, but have gotten stuck on loading Sponza's texture. A brief examination has led me to seeing that some of the textures (1024 x 1024 x 8)  and have been getting jlong_disjoint_arraycopy when trying to create a buffer of some size. Apparently JVM can only handle 2GB, but that should not be the issue here.  Just wondering how you addressed this?

**Comment by lwjglgamedev** [comment]
I will have a look at it. Java buffers are limited because its capacity is limited to maximum int value, therefore the 2GB limit. I will try to check if sopmethnig can be done. In any case isn't STB also limited around 2GB also?

**Comment by lwjglgamedev** [comment]
Ok, Just gave it a quick look. It seems that you could do what is explained [here](http://forum.lwjgl.org/index.php?topic=7141.0), or wait until LWGL 4.

**Reply by Hellwoodoo** [reply]
Hmm thanks, looks helpful. Also on that note - when i tried replicating your array of samplers - am encountering an issue where the samplers appear in descriptor set as 2[0], 2[1] etc, but is not showing up as a resource in the fragment shader during the indirect indexed command. Would you know what a possible issue for this could be? Validation layers are not complaining either :(

**Comment by lwjglgamedev** [comment]
No clue about what could be happening. Are images in the proper layout? What can see in Renderdoc?
